File: src/java/magmac/app/lang/node/Lambda.java
package magmac.app.lang.node;

/**
 * A lambda expression consisting of a header (parameters) and content. For
 * example {@code (x) -> x + 1} would be represented by this node.
 */

import magmac.app.lang.java.JavaLang;

public class Lambda {
    private final JavaLang.JavaLambdaHeader header;
    private final JavaLang.JavaLambdaContent content;

    public Lambda(JavaLang.JavaLambdaHeader header, JavaLang.JavaLambdaContent content) {
        this.header = header;
        this.content = content;
    }

    public JavaLang.JavaLambdaHeader header() {
        return header;
    }

    public JavaLang.JavaLambdaContent content() {
        return content;
    }
}

File: src/web/magmac/app/lang/node/Lambda.ts
import { JavaLang } from "../../../../magmac/app/lang/java/JavaLang";
export class Lambda {
	Lambda(header : JavaLang.JavaLambdaHeader, content : JavaLang.JavaLambdaContent) : public {this.header=header;this.content=content;;}
	header() : JavaLang.JavaLambdaHeader {return header;;}
	content() : JavaLang.JavaLambdaContent {return content;;}
}
